nemo/collections/asr/modules/transformer/transformer_generators.py

```diff
@@ -157,12 +157,13 @@ def _one_step_forward(
                 return_mems=True,
             )
         else:
-            decoder_mems_list = self.decoder.forward(
+            decoder_mems_list, _ = self.decoder.forward(
                 decoder_hidden_states, decoder_input_mask, decoder_mems_list, return_mems=True
             )
         with self.classifier.with_log_softmax_enabled(return_scores) as clf:
             logits = clf.forward(hidden_states=decoder_mems_list[-1][:, -1:])
-        return logits, decoder_mems_list
+
+        return logits, decoder_mems_list, xatt_scores_list
```

Comment:
if `encoder_hidden_states` is None, `xatt_scores_list` is undefined (here will be an error)